ignite/metrics/metric.py

```diff
@@ -108,14 +115,37 @@ def _sync_all_reduce(self, tensor: Union[torch.Tensor, numbers.Number]) -> Union
         else:
             raise TypeError("Unhandled input type {}".format(type(tensor)))
 
-        # synchronize and reduce
-        dist.barrier()
-        dist.all_reduce(tensor)
+        reduction(tensor)
 
         if tensor_to_number:
             return tensor.item()
         return tensor
 
+    @staticmethod
+    def _tpu_reduce(tensor: Union[torch.Tensor, numbers.Number]) -> None:
+        xm.all_reduce("sum", [tensor,])
+
+    @staticmethod
+    def _gpu_reduce(tensor: Union[torch.Tensor, numbers.Number]) -> None:
+        dist.barrier()
+        dist.all_reduce(tensor)
+
+    def _tpu_sync_all_reduce(self, tensor: Union[torch.Tensor, numbers.Number]) -> Union[torch.Tensor, numbers.Number]:
+        return self._do_reduction(tensor, self._tpu_reduce)
+
+    def _gpu_sync_all_reduce(self, tensor: Union[torch.Tensor, numbers.Number]) -> Union[torch.Tensor, numbers.Number]:
+        return self._do_reduction(tensor, self._gpu_reduce)
+
+    def _sync_all_reduce(self, tensor: Union[torch.Tensor, numbers.Number]) -> Union[torch.Tensor, numbers.Number]:
+        if on_xla_device:
```

Comment:
IMO, we need to test something equivalent to `dist.is_initialized()` which can be probably done by `xm.xrt_world_size() > 1`. This I checked on Colab.